crates/bevy_ecs/src/query/state.rs

```diff
@@ -526,9 +516,10 @@ impl<Q: WorldQuery, F: ReadOnlyWorldQuery> QueryState<Q, F> {
     #[inline]
     pub fn iter_mut<'w, 's>(&'s mut self, world: &'w mut World) -> QueryIter<'w, 's, Q, F> {
         // SAFETY: query has unique world access
+        let change_tick = world.change_tick();
         unsafe {
             self.update_archetypes(world);
```

Comment:
I'm not familiar with the code at all but I noticed this method isn't `unsafe`. Is there a reason to call it inside the unsafe block then?

Reply:
It's not unsafe and it could be moved out of the unsafe block. Probably not something to be concerned about for this PR though.

Reply:
No, it's not required, and the code quality benefit isn't too big.